docs/source/developers/python.rst

```diff
@@ -313,24 +313,26 @@ created above (stored in ``$ARROW_HOME``):
 
    $ mkdir arrow/cpp/build
    $ pushd arrow/cpp/build
-
    $ cmake -DCMAKE_INSTALL_PREFIX=$ARROW_HOME \
            -DCMAKE_INSTALL_LIBDIR=lib \
            -DCMAKE_BUILD_TYPE=Debug \
+           -DARROW_BUILD_TESTS=ON \
+           -DARROW_COMPUTE=ON \
+           -DARROW_CSV=ON \
            -DARROW_DATASET=ON \
+           -DARROW_FILESYSTEM=ON \
+           -DARROW_HDFS=ON \
+           -DARROW_JSON=ON \
+           -DARROW_PARQUET=ON \
+           -DARROW_WITH_BROTLI=ON \
            -DARROW_WITH_BZ2=ON \
-           -DARROW_WITH_ZLIB=ON \
-           -DARROW_WITH_ZSTD=ON \
            -DARROW_WITH_LZ4=ON \
            -DARROW_WITH_SNAPPY=ON \
-           -DARROW_WITH_BROTLI=ON \
-           -DARROW_PARQUET=ON \
+           -DARROW_WITH_ZLIB=ON \
+           -DARROW_WITH_ZSTD=ON \
            -DPARQUET_REQUIRE_ENCRYPTION=ON \
-           -DARROW_PYTHON=ON \
-           -DARROW_BUILD_TESTS=ON \
            ..
-   $ make -j4
-   $ make install
+   $ cmake --build . --target install --config Debug
```

Comment:
For my understanding: is the `--config Debug` needed if you already use `CMAKE_BUILD_TYPE=Debug` above? 

And will this line also build in parallel?

Reply:
Yeah the `--config Debug` is redundant.

In my experience, it seems to automatically build in parallel, but you can also [add the option explicitly](https://cmake.org/cmake/help/latest/release/3.12.html#command-line).

Reply:
Ah, sorry. I added this accidentally. I revert this.

I found `-j4` isn't suitable here. We can use `-j$(nproc)` on Linux and `-j$(sysctl -n hw.ncpu)` on macOS. But I thought that it's better that we use Ninja here. And I started rewriting this for Ninja but I stopped it because this pull request isn't for the change. But this change remained.